### ISSUE TITLE
`#` in YAML header causing data to be colorized as if it were a comment

### ISSUE THREAD

**Issue by dethmasque** [rstudio/rstudio#2591]
### System details

    RStudio Edition: Desktop
    RStudio Version: Version 1.1.442
    OS Version:  Mac OS Sierra 10.12.5
    R Version: 3.4.2

### Steps to reproduce the problem

Create an `.Rmd` with the following YAML header:

```
---
rmd_output_metadata:
  rsc_email_subject: i'm an email subject !@#$%^&*()_+~`=
---
```

### Describe the problem in detail

The `#` is causing RStudio to believe that the remainder of `rsc_email_subject` is a comment and is colorizing it as such:

<img width="501" alt="screen shot 2018-04-04 at 3 38 21 pm" src="https://user-images.githubusercontent.com/32644679/38331045-298a24cc-3820-11e8-9c92-96143af7b52c.png">

This does not affect the YAML being parsed appropriately.

**Comment by kevinushey** [comment]
Should be fixed up in the next daily builds -- thanks for the report!

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
